pinot-broker/src/test/java/org/apache/pinot/broker/request/PqlAndCalciteSqlCompatibilityTest.java

```diff
@@ -0,0 +1,123 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+/**
+ * Some tests for the SQL compiler.
+ * Please note that this test will load test resources: `pql_queries.list` and `pql_queries.list` under `pinot-common` module.
+ */
+public class PqlAndCalciteSqlCompatibilityTest {
+
+  private static final Pql2Compiler COMPILER = new Pql2Compiler();
+
+  // OPTIMIZER is used to flatten certain queries with filtering optimization.
+  // The reason is that SQL parser will parse the structure into a binary tree mode.
+  // PQL parser will flat the case of multiple children under AND/OR.
+  // After optimization, both BrokerRequests from PQL and SQL should look the same and be easier to compare.
+  private static final BrokerRequestOptimizer OPTIMIZER = new BrokerRequestOptimizer();
+  private static final Logger LOGGER = LoggerFactory.getLogger(PqlAndCalciteSqlCompatibilityTest.class);
+
+  @Test
+  public void testSinglePqlAndSqlCompatible() {
+    final String sql =
+        "SELECT CarrierDelay, Origin, DayOfWeek FROM mytable WHERE ActualElapsedTime BETWEEN 163 AND 322 OR CarrierDelay IN (17, 266) OR AirlineID IN (19690, 20366) ORDER BY TaxiIn, TailNum LIMIT 1";
+    final String pql =
+        "SELECT CarrierDelay, Origin, DayOfWeek FROM mytable WHERE ActualElapsedTime BETWEEN 163 AND 322 OR CarrierDelay IN (17, 266) OR AirlineID IN (19690, 20366) ORDER BY TaxiIn, TailNum LIMIT 1";
+
+    // PQL
+    LOGGER.info("Trying to compile PQL: {}", pql);
+    // NOTE: SQL is always using upper cases, so we need to make the string to upper case in order to match the parsed identifier name.
+    final BrokerRequest unOptimizedBrokerRequestFromPQL = COMPILER.compileToBrokerRequest(pql);
+    final BrokerRequest brokerRequestFromPQL = OPTIMIZER.optimize(unOptimizedBrokerRequestFromPQL, null);
+    LOGGER.debug("Compiled PQL: PQL: {}, BrokerRequest: {}", pql, brokerRequestFromPQL);
+    brokerRequestFromPQL.unsetPinotQuery();
+
+    //SQL
+    LOGGER.info("Trying to compile SQL: {}", sql);
+    final PinotQuery pinotQuery = CalciteSqlParser.compileToPinotQuery(sql);
+    final BrokerRequest brokerRequestFromSQL =
+        OPTIMIZER.optimize(new PinotQuery2BrokerRequestConverter().convert(pinotQuery), null);
+    LOGGER.debug("Compiled SQL: SQL: {}, PinotQuery: {}, BrokerRequest: {}", sql, pinotQuery, brokerRequestFromSQL);
+
+    // Compare
+    LOGGER.info("Trying to compare BrokerRequest -\nFrom PQL: {}\nFrom SQL: {}", brokerRequestFromPQL,
+        brokerRequestFromSQL);
+    Assert.assertTrue(BrokerRequestComparisonUtils.validate(brokerRequestFromPQL, brokerRequestFromSQL));
+  }
+
+  @Test
+  public void testPqlAndSqlCompatible()
+      throws Exception {
+    final BufferedReader brPql = new BufferedReader(new InputStreamReader(
+        PqlAndCalciteSqlCompatibilityTest.class.getClassLoader().getResourceAsStream("pql_queries.list")));
```

Comment:
Are the queries.list comprehensive enough to test all the various PQL syntax features?

Reply:
I don't think this will cover full syntax features. I've added some more queries into it. We should keep adding more query patterns.

Reply:
Ok, we should ensure that we have full coverage before we advertise the SQL option. Let's open a issue to track that.

Reply:
will do